docker/entrypoint.sh

```diff
@@ -28,8 +29,59 @@ do
   task_exe  $i &
 done
 
+# Function to wait for document engine to be ready
+wait_for_doc_engine() {
+  # Get document engine type
+  DOC_ENGINE=${DOC_ENGINE:-elasticsearch}
+  echo "Using document engine: $DOC_ENGINE"
+
+  if [ "$DOC_ENGINE" = "elasticsearch" ]; then
+    # Get Elasticsearch configuration
+    ES_HOST=${ES_HOST:-es01}
+    ES_PORT=${ES_PORT:-1200}
+    ELASTIC_PASSWORD=${ELASTIC_PASSWORD:-infini_rag_flow}
+    
+    echo "Waiting for Elasticsearch (${ES_HOST}:${ES_PORT}) to be ready..."
+    
+    # Build health check URL
+    ES_URL="http://${ES_HOST}:${ES_PORT}"
+    
+    # Add authentication header
+    AUTH_HEADER=""
+    if [ -n "$ELASTIC_PASSWORD" ]; then
+      AUTH_HEADER="-u elastic:${ELASTIC_PASSWORD}"
+    fi
+    
+    until curl -s -f $AUTH_HEADER "${ES_URL}/_cluster/health?wait_for_status=yellow" > /dev/null 2>&1; do
+      echo "Elasticsearch is not ready - waiting"
+      sleep 5
+    done
+    echo "Elasticsearch is ready - starting RAGFlow"
+  elif [ "$DOC_ENGINE" = "infinity" ]; then
+    # Get Infinity configuration
+    INFINITY_HOST=${INFINITY_HOST:-infinity}
+    INFINITY_HTTP_PORT=${INFINITY_HTTP_PORT:-23820}
+    
+    echo "Waiting for Infinity (${INFINITY_HOST}:${INFINITY_HTTP_PORT}) to be ready..."
+    
+    # Build health check URL
+    INFINITY_URL="http://${INFINITY_HOST}:${INFINITY_HTTP_PORT}"
+    
+    until curl -s -f "${INFINITY_URL}/admin/node/current" > /dev/null 2>&1; do
+      echo "Infinity is not ready - waiting"
+      sleep 5
+    done
+    echo "Infinity is ready - starting RAGFlow"
+  fi
+}
+
+# Call the function
+wait_for_doc_engine
```

Comment:
`ESConnection.__init__`, `InfinityConnection.__init__` wait till the doc engine be ready.